bottles/frontend/windows/executesettings.py

```diff
@@ -36,17 +36,10 @@ def __init__(self, parent, **kwargs):
         self.parent = parent
 
         # connect signals
-        self.btn_run.connect("clicked", self.__run_executable)
-        self.btn_cancel.connect("clicked", self.__close_window)
-
-    def __run_executable(self, widget):
-        """
-        This function return the user to the executable selection with
-        the new typed arguments, then close the window.
-        """
+        self.btn_done.connect("clicked", self.__done)
+
+    def __done(self, widget):
         args = self.entry_args.get_text()
         self.parent.run_executable(False, args)
```

Comment:
One odd behavior I noticed is that pressing "Done" opens a file chooser dialog for picking an executable. It runs the `run_executable()` function from `bottle_preferences.py`, which is the function that opens the file chooser dialog. I propose to make it close the dialog so the user can manually press "Run Executable" and launch with the options they set. Beware, the "Run Executable" button runs the `run_executable()` function as well, which needs to use the file chooser dialog. Here's a video that demonstrates the behavior:

https://user-images.githubusercontent.com/50847364/197288595-1eca8473-2619-48d5-b18f-e7ca48635276.mp4

Another odd behavior is that the settings aren't saved in the session. What it should do is keep those settings until the app is restarted. Here's the video that demonstrates the bug:

https://user-images.githubusercontent.com/50847364/197290970-5e9d446a-7a91-423d-8e0e-276479513a5f.mp4